picture-stealer/src/main/java/com/bobocode/picturestealer/service/CameraService.java

```diff
@@ -0,0 +1,44 @@
+package com.bobocode.picturestealer.service;
+
+import com.bobocode.picturestealer.dto.Camera;
+import com.bobocode.picturestealer.entity.CameraEntity;
+import com.bobocode.picturestealer.manager.CameraRepository;
+import jakarta.persistence.LockModeType;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.jpa.repository.Lock;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * @author "Maksym Oliinyk"
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CameraService {
+
+    private final CameraRepository cameraRepository;
+
+    @Lock(LockModeType.OPTIMISTIC_FORCE_INCREMENT)
+    @Transactional(propagation = Propagation.REQUIRED)
+    public CameraEntity findOrCreateCamera(Camera camera) {
+        return cameraRepository.findByNasaId(camera.id())
```

Comment:
Minor (code style). In Effective Java, Joshua Block says that it's generally a bad practice to write lambda statements (blocks of code with curly braces). It's not critical, but it looks a bit ugly to me.

Reply:
@tboychuk I'm afraid I have to disagree with you, using curly brackets in lambda expressions is not inherently a bad practice; rather, it's a matter of readability and conciseness based on the context. Joshua Bloch writes next: "One line is ideal for a lambda, and three lines is a reasonable maximum. If you violate this rule, it can cause serious harm to the readability of your programs."